lib/ronin/db/http_request.rb

```diff
@@ -115,6 +116,17 @@ class HTTPRequest < ActiveRecord::Base
       #   @return [Time]
       attribute :created_at, :datetime
 
+      # @!attribute [rw] address
+      #   The source IP Address.
+      #
+      #   @return [String]
+      attribute :source_ip, :string
+      validates :source_ip, length: { maximum: 39 },
+                            format: {
+                              with: /#{Resolv::IPv4::Regex}|#{Resolv::IPv6::Regex}/,
+                              message: 'Must be a valid IP address'
+                            },
+                            allow_blank: true
```

Comment:
Use `allow_nil: true` instead of `allow_blank: true`. We want to allow `nil`, but not empty-strings.